src/main/kotlin/inc/evil/d17map/security/config/CustomAccessDeniedHandler.kt

```diff
@@ -0,0 +1,26 @@
+package inc.evil.d17map.security.config
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.security.access.AccessDeniedException
+import org.springframework.security.web.access.AccessDeniedHandler
+import org.springframework.stereotype.Component
+
+@Component
+class CustomAccessDeniedHandler : AccessDeniedHandler {
+    override fun handle(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        accessDeniedException: AccessDeniedException
+    ) {
+        response.contentType = "application/json"
+        response.status = HttpServletResponse.SC_FORBIDDEN
+
+        val errorDetails: MutableMap<String, Any> = HashMap()
+        errorDetails["message"] = "Forbidden: " + accessDeniedException.message
```

Comment:
````
val errorDetails = mutableMapOf("message" to "Forbidden: ${accessDeniedException.message}")

Reply:
Można też dodać jakieś dodatkowe informacje, żeby w przyszłości jak wystąpi jakiś błąd łatwiej było to debugować 

`````
errorDetails["errorCode"] = "ACCESS_DENIED"
errorDetails["timestamp"] = Instant.now().toString()

Reply:
Done